include/odp/api/spec/proto_stats.h

```diff
@@ -27,6 +27,25 @@ extern "C" {
 
 /** ODP proto stats id */
 typedef enum odp_proto_stats_id_t {
+	/** Tx proto stats id ODP_PROTO_STATS_ID_TX_*
+	 *
+	 * - Statistics that can be enabled in proto stats objects for Tx of a packet via
+	 *   Packet IO. Packet IO pktout config `odp_pktout_config_opt_t::bit::proto_stats_ena`
+	 *   needs to be enabled forthe offload to work.
+	 */
+
+	/** Packet sent count */
+	ODP_PROTO_STATS_ID_TX_PKT,
+	/** Packet drop count */
+	ODP_PROTO_STATS_ID_TX_PKT_DROP,
```

Comment:
I started wondering if the API is too ambiguous regarding what packet drops update the counter. My (maybe unfounded) assumption has been that the drop counter gets incremented if the packet gets dropped in the output path for any reason, including TM and IPsec. In other words, whenever the sent counter is not increased, the drop count is. But I am not sure if the API actually says that or whether that is even the intention.

Even if we wanted to count all drops regardless of where they occur, I am not sure if inline outbound IPsec really drops many packets as opposed to returning them to the application through an error event.

Reply:
I agree. In our platform, ODP_PROTO_STATS_ID_TX_PKT_DROP will be counting only packets either dropped due to TM shaping/ratelimiting or packet Aging. It doesn't count drops due to Outbound IPSec processing or other cases.

But can't it be left this way as it is really platform architecture dependent ?

Reply:
Updated the text to clearly state that what all kind of Tx pkt statistics are counted is implementation defined in next version. I hope it is fine.